sigoptlite/sigoptlitetest/endpoints/test_best_assignments.py

```diff
@@ -127,108 +163,146 @@ def test_multimetric_thresholds(self, threshold_y1, threshold_y2):
+    assert best_observation.values[0].value >= threshold
 
-    best_observation = best_observations_from_logger[0]
-    assert best_observation["values"][0]["value"] >= threshold
+    observations = self.conn.experiments(e.id).observations().fetch()
+    observations_list = list(observations.iterate_pages())
+    optimized_values_within_threshold = []
+    for observation in observations_list:
+      if observation.values[0].value >= threshold:
+        optimized_values_within_threshold.append(observation.values[1].value)
 
-    valid_observations = best_assignments_logger.filter_valid_full_cost_observations(observations)
-    expected_valid_observations = [o.get_client_observation(experiment) for o in valid_observations]
-    for observation in expected_valid_observations:
-      assert best_observation["values"][1]["value"] <= observation["values"][1]["value"]
+    for optimized_value in optimized_values_within_threshold:
+      assert best_observation.values[1].value <= optimized_value
 
   def test_multimetric_search(self):
     experiment_meta = self.get_experiment_feature("search")
-    thresholds = [m["threshold"] for m in experiment_meta["metrics"]]
-    objectives = [m["objective"] for m in experiment_meta["metrics"]]
-    experiment = LocalExperimentBuilder(experiment_meta)
-    observations = self.make_random_observations(experiment, num_observations=25)
-
-    best_assignments_logger = BestAssignmentsLogger(experiment)
-    best_observations_from_logger = best_assignments_logger.fetch(observations)
-    self.assert_observations_are_sorted(best_observations_from_logger, experiment)
-
-    for best_observation in best_observations_from_logger:
-      for i, metric in enumerate(best_observation["values"]):
-        if objectives[i] == "maximize":
-          assert metric["value"] >= thresholds[i]
-        if objectives[i] == "minimize":
-          assert metric["value"] <= thresholds[i]
-
-    valid_observations = best_assignments_logger.filter_valid_full_cost_observations(observations)
-    expected_valid_observations = [o.get_client_observation(experiment) for o in valid_observations]
-    assert len(best_observations_from_logger) > 1
-    self.assert_observation_lists_are_equal(best_observations_from_logger, expected_valid_observations)
+    experiment_meta["type"] = "random"
+    e = self.conn.experiments().create(**experiment_meta)
+
+    num_observations = 25
+    for _ in range(num_observations):
+      suggestion = self.conn.experiments(e.id).suggestions().create()
+      self.conn.experiments(e.id).observations().create(
+        suggestion=suggestion.id,
+        values=[dict(name="y1", value=numpy.random.rand()), dict(name="y2", value=numpy.random.rand())],
+      )
+
+    best_assignments = self.conn.experiments(e.id).best_assignments().fetch()
+    best_assignments_list = list(best_assignments.iterate_pages())
+    self.assert_best_assignments_are_sorted(e, best_assignments_list)
+    assert len(best_assignments_list) > 1
```

Comment:
this is probably fine, the chance of this being 0 or 1 is like (0.75 * 0.25)^n  or (n-1).

Reply:
yea that's why I jacked up the number of observations